### DISCUSSION TITLE
Yarn@v4 support for non-static builds

### DISCUSSION THREAD

**Discussion by kaisermann** [vercel/community#5664] [Feature Question]
### Question

> I didn't find another discussion about this, sorry if it's a duplicate.

Hey there 👋,

[I'm aware that `yarn@v2` is currently not supported for non-static builds](https://vercel.com/guides/does-vercel-support-yarn-2). Since `yarn@v4` is a big change from v2 or even v3, I would like to know if it's planned to support the v4 version for building SSR apps. At least with the [`nodeLinker: node_modules`](https://yarnpkg.com/configuration/yarnrc#nodeLinker).

**Comment by thclark** [comment]
Even just for static builds it would be a big step up. It's worth pointing out that **Yarn v1 is deprecated** and bugs like [this one](https://github.com/storybookjs/storybook/issues/22431#issuecomment-1630086092) are going to become increasingly more common. I updated my yarn version today without even imagining that Vercel wouldn't support current yarn releases. 

At least offer a statement about it (eg "we're never going to do this because of xyz reason, use pnpm instead")

**Comment by kaisermann** [comment]
For anyone landing here, I did manage to make `yarn@v4` work in Vercel's pipeline. There's a `ENABLE_EXPERIMENTAL_COREPACK` env flag that can see to `1` and it will enable corepack during the CI setup, which in turn supports having yarn@v4 as the package manager.

Related post about the flag: https://vercel.com/changelog/corepack-experimental-is-now-available

**Reply by 3runo** [reply]
Can you share part of your package.json?
Vercel is logging `WARN! Warning: Could not enable corepack because package.json is missing "packageManager" property` in their Deployment Details
